### PR TITLE
Add support fpga drv and pimutility for minipack platform

### PR DESCRIPTION
**Pull request by jostar-yang**
**- What I did**
Add fpga drv and pimutility code
**- How I did it**
Add fpga drv and pimutility code
**- How to verify it**
Set / get sfp presence from pim.
Set/ get lpmode from pim
root@sonic:/home/admin# sfputil show presence

Port         Presence
-----------  -----------
Ethernet1    Present
Ethernet2    Present
Ethernet3    Present
Ethernet4    Present
Ethernet5    Present
Ethernet6    Not present
Ethernet7    Not present
Ethernet8    Not present
........

sfputil lpmode off Ethernet1

sfputil lpmode off Ethernet2

root@sonic:/home/admin# sfputil show lpmode
Port         Low-power Mode
-----------  ----------------
Ethernet1    Off
Ethernet2    Off
Ethernet3    On
Ethernet4    On
Ethernet5    On
Ethernet6    On
Ethernet7    On
.......
**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by lguohan** [comment]
retest vs please

File: device/accton/x86_64-accton_minipack-r0/plugins/sfputil.py [hunk 1]
```diff
@@ -61,34 +64,96 @@ def __init__(self):
...
+        status=pim.get_qsfp_presence(port_num)
+        return status
+        #eeprom_path = self.port_to_eeprom_mapping[port_num]
+        #with open(eeprom_path) as f:
+        #    try:
+        #        content = f.read(1)
+        #    except IOError as e:
+        #        #Not print any error, for if any, treat as Not present.
+        #        return False
+        #return True
```
Comment (lguohan) [inline device/accton/x86_64-accton_minipack-r0/plugins/sfputil.py]: suggest to remove commented code

File: device/accton/x86_64-accton_minipack-r0/plugins/sfputil.py [hunk 2]
```diff
@@ -61,34 +64,96 @@ def __init__(self):
...
     def reset(self, port_num):
-        raise NotImplementedError
+        if port_num < self.port_start or port_num > self.port_end:
+            return False
+        pim=PimUtil()
+        pim.reset(port_num)
+        return True
+        
+    #return code=" port_dict[port]='1' ":insert evt. return code="port_dict[port]='0' ":remove evt
+    def get_transceiver_change_event(self, timeout=0):
```
Comment (lguohan) [inline device/accton/x86_64-accton_minipack-r0/plugins/sfputil.py]: remove commented code

File: platform/broadcom/platform-modules-accton.mk [hunk 3]
```diff
@@ -83,4 +83,5 @@ ACCTON_AS5835_54X_PLATFORM_MODULE = sonic-platform-accton-as5835-54x_$(ACCTON_AS
 $(ACCTON_AS5835_54X_PLATFORM_MODULE)_PLATFORM = x86_64-accton_as5835_54x-r0
 $(eval $(call add_extra_package,$(ACCTON_AS7712_32X_PLATFORM_MODULE),$(ACCTON_AS5835_54X_PLATFORM_MODULE)))
 
+
```
Comment (lguohan) [inline platform/broadcom/platform-modules-accton.mk]: remove empty blank line.

File: platform/broadcom/sonic-platform-modules-accton/debian/control [hunk 4]
```diff
@@ -56,3 +56,4 @@ Description: kernel modules for platform devices such as fan, led, sfp
 Package: sonic-platform-accton-as5835-54x
 Architecture: amd64
 Description: kernel modules for platform devices such as fan, led, sfp
+
```
Comment (lguohan) [inline platform/broadcom/sonic-platform-modules-accton/debian/control]: remove blank line

File: platform/broadcom/sonic-platform-modules-accton/debian/rules [hunk 5]
```diff
@@ -19,7 +19,7 @@ PACKAGE_PRE_NAME := sonic-platform-accton
 KVERSION   ?= $(shell uname -r)
 KERNEL_SRC :=  /lib/modules/$(KVERSION)
 MOD_SRC_DIR:= $(shell pwd)
-MODULE_DIRS:= as7712-32x as5712-54x as7816-64x as7716-32x as7716-32xb as7312-54x as7326-56x as6712-32x as7726-32x as4630-54pe minipack as5812-54x as5835-54x
+MODULE_DIRS:= as7712-32x as5712-54x as7816-64x as7716-32x as7716-32xb as7312-54x as7326-56x as6712-32x as7726-32x as4630-54pe minipack as5812-54x as5835-54x
```
Comment (lguohan) [inline platform/broadcom/sonic-platform-modules-accton/debian/rules]: remove extra space at the end of line

File: platform/broadcom/sonic-platform-modules-accton/minipack/classes/pimutil.py [hunk 6]
```diff
@@ -0,0 +1,575 @@
...
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (lguohan) [inline platform/broadcom/sonic-platform-modules-accton/minipack/classes/pimutil.py]: user space program should apache 2 license, not gpl.

**Comment by lguohan** [comment]
please add [devices]:  Add support fpga drv and pimutility for [device name]

File: device/accton/x86_64-accton_minipack-r0/plugins/sfputil.py (hunk 2 above)
Comment (jostar-yang) [reply device/accton/x86_64-accton_minipack-r0/plugins/sfputil.py]: Done

File: device/accton/x86_64-accton_minipack-r0/plugins/sfputil.py (hunk 1 above)
Comment (jostar-yang) [reply device/accton/x86_64-accton_minipack-r0/plugins/sfputil.py]: done

File: platform/broadcom/platform-modules-accton.mk (hunk 3 above)
Comment (jostar-yang) [reply platform/broadcom/platform-modules-accton.mk]: Done

File: platform/broadcom/sonic-platform-modules-accton/debian/control (hunk 4 above)
Comment (jostar-yang) [reply platform/broadcom/sonic-platform-modules-accton/debian/control]: Done

File: platform/broadcom/sonic-platform-modules-accton/debian/rules (hunk 5 above)
Comment (jostar-yang) [reply platform/broadcom/sonic-platform-modules-accton/debian/rules]: Done

File: platform/broadcom/sonic-platform-modules-accton/minipack/classes/pimutil.py (hunk 6 above)
Comment (jostar-yang) [reply platform/broadcom/sonic-platform-modules-accton/minipack/classes/pimutil.py]: Done

Comment (jostar-yang) [reply platform/broadcom/sonic-platform-modules-accton/minipack/classes/pimutil.py]: changed to apache 2 license

**Comment by lguohan** [comment]
retest vsimage please

**Pull request merged**